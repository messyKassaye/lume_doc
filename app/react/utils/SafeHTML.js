import { Parser } from 'html-to-react';
import PropTypes from 'prop-types';
import React from 'react';

const myParser = new Parser();
const stringToReact = string => myParser.parse(string);

const SafeHTML = ({ children }) => <>{stringToReact(children)}</>;

SafeHTML.propTypes = {
  children: PropTypes.string.isRequired,
};

export default SafeHTML;
