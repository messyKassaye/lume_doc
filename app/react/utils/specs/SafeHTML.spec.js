import React from 'react';

import { shallow } from 'enzyme';

import SafeHTML from '../SafeHTML';

describe('SafeHTML', () => {
  it('should transform <snippet/> tag to <b> (every other tag should be treated as text)', () => {
    const component = shallow(
      <SafeHTML>{'text with <span>some</span> <b>html</b> tags'}</SafeHTML>
    );
    expect(component).toMatchSnapshot();
  });
});
