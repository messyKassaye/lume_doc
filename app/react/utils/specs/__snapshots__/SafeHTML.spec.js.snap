// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeHTML should transform <snippet/> tag to <b> (every other tag should be treated as text) 1`] = `
<Fragment>
  text with 
  <span
    key="1"
  >
    some
  </span>
   
  <b
    key="3"
  >
    html
  </b>
   tags
</Fragment>
`;
